src/chat.rs

```diff
@@ -592,17 +608,31 @@ impl ChatId {
         // the times are average, no matter if there are fresh messages or not -
         // and have to be multiplied by the number of items shown at once on the chatlist,
         // so savings up to 2 seconds are possible on older devices - newer ones will feel "snappier" :)
-        let count = context
-            .sql
-            .count(
-                "SELECT COUNT(*)
+        let count = if self.is_deaddrop() {
+            context
+                .sql
+                .count(
+                    "SELECT COUNT(*)
+                    FROM msgs
+                    WHERE state=?
+                    AND hidden=0
+                    AND chat_id IN (SELECT id FROM chats WHERE blocked=?)",
```

Comment:
at `get_chat_msgs()` we are using a LEFT JOIN instead of a subselect - would be interesting, which one is faster, esp. as this, once introduced for deaddrops may be a function frequently called.

(for non-deaddrops we also did some optimisations, see comment above)

however, if needed, that is sth. for another pr.